andela/matterhorn-cfh
# #152024522 Show Donor Identity

### What does this PR do?
Displays donors who gives positive consent to show their identity, before donating on crowdrise

### Description of Task to be completed?
A modal is shown when `Make Donation` button is clicked, Prompting user to give consent on allowing their identity shown as donors, if response is Yes, this is temporarily stored in the `localStorage` and on call back of the request from **crowdrise** the temporarily saved response is retrieved and passed along with the post data sent to the database for the user.

An API call is then made to get all users, their donations are then gotten where the `donor_consent` field is `true` this is stored in a `$scope.donors` array made available in a materialize carousel, which is displayed on the **homepage**

### How should this be manually tested?
- Pull in latest changes
- Install all dependencies after pulling `npm install` or `npm update`
- run `npm build` then run `npm start`
- Log in to the application
- Be Nice, make a donation (at least you'll help test if callback is working as expected)
- Visit home page after donation is made, you should see a carousel with _donor name_ and the _amount donated_

### Any background context you want to provide?
- Donations must be made and positive (**Yes**) consent given before it can show on the list
- The Donations carousel can be viewed by all users (both guest and authenticated users)

### What are the relevant pivotal tracker stories?
#152024522 Users should get a consent form to show their identity as a donor

### Screenshots (if appropriate)
<img width="1437" alt="screen shot 2017-10-18 at 11 36 26 pm" src="https://user-images.githubusercontent.com/20375577/31746070-70af180e-b45d-11e7-8396-b1035c21ffca.png">

<img width="1438" alt="screen shot 2017-10-18 at 11 36 41 pm" src="https://user-images.githubusercontent.com/20375577/31746073-7ae19590-b45d-11e7-9613-0daf657619cd.png">

### Questions:
N/A
